torch/csrc/distributed/c10d/NCCLUtils.hpp

```diff
@@ -235,6 +240,22 @@ class NCCLComm {
   }
 #endif
 
+#ifdef NCCL_HAS_COMM_SPLIT
+  static std::shared_ptr<NCCLComm> split(
+      NCCLComm* source,
+      int color_id,
+      int rank,
+      ncclConfig_t& config) {
+    auto comm = std::make_shared<NCCLComm>();
+    C10D_NCCL_CHECK(
+        ncclCommSplit(
+            source->ncclComm_, color_id, rank, &(comm->ncclComm_), &config),
+        c10::nullopt);
```

Comment:
nit: maybe clearer if `c10::nullopt` is replaced by an error message?

Reply:
I think `C10D_NCCL_CHECK` will build a nice contextual error message when this is null and this seems consistent with what it does on the other call sites.